crates/telio-sockets/src/protector/linux.rs

```diff
@@ -61,3 +61,48 @@ fn set_fwmark(fd: i32, fwmark: u32) -> io::Result<()> {
         _ => Err(io::Error::last_os_error()),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{
+        io::ErrorKind,
+        net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr, TcpListener, UdpSocket},
+    };
+
+    use crate::native::AsNativeSocket;
+
+    use rstest::rstest;
+
+    use super::*;
+
+    #[rstest]
+    #[case(IpAddr::V4(Ipv4Addr::LOCALHOST))]
+    #[case(IpAddr::V4(Ipv4Addr::UNSPECIFIED))]
+    #[case(IpAddr::V6(Ipv6Addr::LOCALHOST))]
+    #[case(IpAddr::V6(Ipv6Addr::UNSPECIFIED))]
+    fn test_make_external(#[case] ip_addr: IpAddr) {
+        let protector = NativeProtector::new().unwrap();
+        protector.set_fwmark(11673110);
+        let addr = SocketAddr::new(ip_addr, 0);
+
+        let socket = match UdpSocket::bind(addr) {
+            Ok(socket) => socket,
+            Err(e) if e.kind() == ErrorKind::AddrNotAvailable => {
+                // host has no interface with ipv4/ipv6
+                return;
+            }
+            Err(e) => panic!("Unexpected error: {:?}", e),
+        };
+        assert!(protector.make_external(socket.as_native_socket()).is_ok());
+
+        let socket = match TcpListener::bind(addr) {
+            Ok(socket) => socket,
+            Err(e) if e.kind() == ErrorKind::AddrNotAvailable => {
+                // host has no interface with ipv4/ipv6
+                return;
```

Comment:
This might hide issues if CI/CD would start failing with this error.

Reply:
What kind of situation do you have in mind? I think the error can only happen the ipv4 or ipv6 interface is not present, see: https://doc.rust-lang.org/stable/std/io/enum.ErrorKind.html#variant.AddrNotAvailable